language/move-lang/src/parser/syntax.rs

```diff
@@ -675,13 +675,26 @@ fn parse_call_args<'input>(tokens: &mut Lexer<'input>) -> Result<Spanned<Vec<Exp
     Ok(spanned(tokens.file_name(), start_loc, end_loc, args))
 }
 
+// Return true if the current token is one that might occur after an Exp.
+// This is needed, for example, to check for the optional Exp argument to
+// a return (where "return" is itself an Exp).
+fn at_end_of_exp<'input>(tokens: &mut Lexer<'input>) -> bool {
+    match tokens.peek() {
+        // These are the tokens that can occur after an Exp. If the grammar
+        // changes, we need to make sure that these are kept up to date and that
+        // none of these tokens can occur at the beginning of an Exp.
+        Tok::Else | Tok::RBrace | Tok::RParen | Tok::Comma | Tok::Colon | Tok::Semicolon => true,
+        _ => false,
```

Comment:
Nit, thoughts on not using an underscore here? It might make it a bit less brittle?

Reply:
I don't think that helps -- the most likely cause of fragility would be changing the grammar, not adding new tokens -- and there are a lot of other tokens that would need to be listed.